src/distance-along-lookup-helper.js

```diff
@@ -0,0 +1,30 @@
+async function getDistanceAlongLookup(parsedStopTimesReadStream) {
+  let rowIdx = 0;
+  let stopIdIdx = 0;
+  let tripIdIdx = 0;
+  let distanceAlongIdx = 0;
+  const distanceAlongLookup = {};
+  for await (const stopCsv of parsedStopTimesReadStream) {
+    if (rowIdx === 0) {
+      stopIdIdx = stopCsv.indexOf('stop_id');
+      tripIdIdx = stopCsv.indexOf('trip_id');
+      distanceAlongIdx = stopCsv.indexOf('shape_dist_traveled');
```

Comment:
Pull this out of the loop then loop over rows 1+. You can get rid of `rowIdx` and simplify. `Col` might be clearer than `Idx`, as it's not something iterated over, e.g. `stopIdCol`

Reply:
not a `[Symbol.asyncIterator]` expert, but as far as I can tell, there isn't an `iterator.next()` equivalent for asyncIterators, the `for await .. of` syntax feels like a special snowflake.

My source:
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/for-await...of

This seems like the cleanest way to do it, the other option is callback hell with Stream events.

Reply:
How about using the [`columns: true` option](https://csv.js.org/parse/options/columns/) and avoid having to deal with indices at all?

Reply:
oh, yea thats a good find